integration-tests/tests/version_management.rs

```diff
@@ -0,0 +1,116 @@
+mod common;
+
+use common::*;
+
+#[tokio::test]
+async fn test_version_stage_transitions() {
+    let secrets = TestSecrets::setup().await;
+    let secret_name = secrets.secret_name(SecretType::Versioned);
+
+    let agent = AgentProcess::start().await;
+
+    // Wait for AWSPENDING version to be available
+    let _ = secrets
+        .wait_for_pending_version(SecretType::Versioned)
+        .await;
+
+    // Get the version IDs for both stages
+    let (current_version_id, pending_version_id) =
+        secrets.get_version_ids(SecretType::Versioned).await;
+
+    // Test AWSPENDING stage before promotion
+    let pending_query = AgentQueryBuilder::default()
+        .secret_id(&secret_name)
+        .version_stage("AWSPENDING")
+        .build()
+        .unwrap();
+    let pending_response = agent.make_request(&pending_query).await;
+    let pending_json: serde_json::Value = serde_json::from_str(&pending_response).unwrap();
+
+    assert_eq!(pending_json["VersionId"], pending_version_id);
+    assert!(pending_json["SecretString"]
+        .as_str()
+        .unwrap()
+        .contains("pendinguser"));
+
+    // Test AWSCURRENT stage before promotion
+    let current_query = AgentQueryBuilder::default()
+        .secret_id(&secret_name)
+        .version_stage("AWSCURRENT")
+        .build()
+        .unwrap();
+    let current_response = agent.make_request(&current_query).await;
+    let current_json: serde_json::Value = serde_json::from_str(&current_response).unwrap();
+
+    assert_eq!(current_json["VersionId"], current_version_id);
+    assert!(current_json["SecretString"]
+        .as_str()
+        .unwrap()
+        .contains("currentuser"));
+
+    // Promote AWSPENDING to AWSCURRENT using update_secret_version_stage
+    let config = aws_config::load_defaults(aws_config::BehaviorVersion::latest()).await;
+    let client = aws_sdk_secretsmanager::Client::new(&config);
+
+    client
+        .update_secret_version_stage()
+        .secret_id(&secret_name)
+        .version_stage("AWSCURRENT")
+        .move_to_version_id(&pending_version_id)
+        .remove_from_version_id(&current_version_id)
+        .send()
+        .await
+        .expect("Failed to promote version stage");
+
+    // Verify promotion worked by checking AWS directly and get updated version IDs
+    let (new_current_version_id, _) = secrets.get_version_ids(SecretType::Versioned).await;
+    assert_eq!(
+        new_current_version_id, pending_version_id,
+        "Version stage promotion failed in AWS"
+    );
+
+    // Small delay to ensure AWS propagation
+    tokio::time::sleep(std::time::Duration::from_millis(500)).await;
```

Comment:
You can do these steps in a waiter similar to [this one](https://github.com/aws/aws-secretsmanager-agent/blob/main/integration-tests/tests/common.rs#L272). This way, you can differentiate between eventual consistency delays in AWS Secrets Manager and failures in the agent retrieval.

Reply:
Let's do this in a separate PR. This is fine.